### ISSUE TITLE
pnpm and prisma client

### ISSUE THREAD

**Issue by mabilbao** [keystonejs/keystone#8975]
Hey guys! 🙌

I installed the project from scratch using npm and yarn and everything works well. But when I try to do the same with pnpm, I get some issues with the folder `.prisma` after removing the `node_modules` and trying to regenerate it.

# Steps to reproduce:
```bash
pnpm dlx create-keystone-app@latest my-app
cd my-app 
pnpm dev # It's working well because "create-keystone-app@latest" used yarn and ./node_modules/.prisma is there 👍
```
![Screenshot 2023-12-22 at 11 28 52](https://github.com/keystonejs/keystone/assets/9256107/8da21aba-dde7-435c-83ba-832712562d9e)

BUT
```
rm -r node_modules
pnpm install
pnpm dev # It's not working due to "Error: Cannot find module '.prisma/client/index'". Of course, ./node_modules/.prisma is not there.
```
![Screenshot 2023-12-22 at 11 31 11](https://github.com/keystonejs/keystone/assets/9256107/28b9626c-b765-4cf5-8f21-c6c7b49e27c7)

---

If I run `pnpm keystone prisma generate`, I get this:

![Screenshot 2023-12-22 at 11 14 49](https://github.com/keystonejs/keystone/assets/9256107/92c387a4-2464-4513-9450-19e1e8f53b57)

And here, 2 things call my attention:
1. Prisma Client is being generated but here: `/node_modules/.pnpm/@prisma+client@5.7.1_prisma@5.7.1/node_modules/@prisma/client`. If I set the code below, it works.
```ts
export default withAuth(
  config({
    db: {
      // we're using sqlite for the fastest startup experience
      //   for more information on what database might be appropriate for you
      //   see https://keystonejs.com/docs/guides/choosing-a-database#title
      provider: 'sqlite',
      url: 'file:./keystone.db',
      prismaClientPath: './node_modules/.pnpm/@prisma+client@5.7.1_prisma@5.7.1/node_modules/.prisma/client'
    },
    lists,
    session,
  })
);
```
2. The error that mention the incompatibility between `@prisma/client` and `prisma`. Am I doing something wrong here?
```
warn Versions of prisma@4.16.2 and @prisma/client@5.7.1 don't match.
This might lead to unexpected behavior.
```

# Versions:
![Screenshot 2023-12-22 at 11 18 09](https://github.com/keystonejs/keystone/assets/9256107/ea4b5437-3ab0-4a0e-af56-ff9ad20924b3)

## `package.json`
```json
{
  "name": "keystone-app",
  "version": "1.0.2",
  "private": true,
  "scripts": {
    "dev": "keystone dev",
    "start": "keystone start",
    "build": "keystone build",
    "postinstall": "keystone build --no-ui --frozen"
  },
  "dependencies": {
    "@keystone-6/auth": "^7.0.0",
    "@keystone-6/core": "^5.0.0",
    "@keystone-6/fields-document": "^7.0.0",
    "@prisma/client": "^5.7.1",
    "typescript": "^4.9.5"
  },
  "devDependencies": {
    "prisma": "^5.7.1"
  }
}

```

Thank you guys!

**Comment by marekryb** [comment]
Hi,

Please see https://github.com/keystonejs/keystone/discussions/8345#discussioncomment-5135749.

**Comment by mabilbao** [comment]
Hi @marekryb 

Thanks for your answer! 😄

1. About the comment on the question: https://github.com/keystonejs/keystone/discussions/8345#discussioncomment-5135749

> In PR https://github.com/keystonejs/keystone/pull/8307 we removed the assumption about where the Prisma Client is generated and added a db.prismaClientPath to enable the configuration of both where Keystone/Prisma will generate the client and where it will import it from.

In this case, when I set `prismaClientPath: 'node_modules/.prisma/client',`, I get `Your Prisma schema is not up to date` after running `pnpm install`. I suppose this is because It is taking that value like the place where Keystone will import the Prisma Client but not like the place where it should generate the client.

Do you know if I'm doing something wrong here?

---

2. It looks like the dependency: `"@prisma/client": "^5.7.1"` is added when I install the project using `pnpm dlx create-keystone-app@latest my-app`. Even If I install the project with npm or yarn, I remove the `node_modules` and run `pnpm install`, the same happens.

`resolutions` fixed my issue! Setting version 4.16.2 or 5.7.1, in both cases it's working. For some reason, `.prisma` is not on the root of `node_modules` but it works anyway (I don't know how honestly).

![Screenshot 2023-12-26 at 12 12 09](https://github.com/keystonejs/keystone/assets/9256107/0f4e501f-f26d-4fdd-8414-3a831a210bd4)

---

Now I have another issue, I have to install this in a Turbo Monorepo and I already have another project with an old version of Keystone. Unfortunately, It looks like I can not set `resolutions` on a child project of the Monorepo, I have to do that in the root, but I will take a look at how can I fix that 🙌

Thanks again!

**Comment by marekryb** [comment]
> In this case, when I set `prismaClientPath: 'node_modules/.prisma/client',`, I get `Your Prisma schema is not up to date` after running `pnpm install`. I suppose this is because It is taking that value like the place where Keystone will import the Prisma Client but not like the place where it should generate the client.
> 
> Do you know if I'm doing something wrong here?

Changing `prismaClientPath` should result in change in `schema.prisma` (`output = "<your path>"` line), so I am guessing that's it. Does it happen even after that?
I vaguely remember having similar issue and the solution was to run `prisma generate` manually.

> Now I have another issue, I have to install this in a Turbo Monorepo and I already have another project with an old version of Keystone. Unfortunately, It looks like I can not set `resolutions` on a child project of the Monorepo, I have to do that in the root, but I will take a look at how can I fix that 🙌

As far as I know monorepo-aware package managers (yarn, pnpm) only allow resultions to be in the root package.json. In your situation keeping prisma in sync with keystone (`prisma@4.16.2`) might be the only solution (other than upgrading your other keystone project).

**Comment by mabilbao** [comment]
Hey @marekryb !

It was my fault. Effectively, it says `Your Prisma schema is not up to date` after `pnpm install` but it's not something bad, all the dependencies are properly installed and `prismaClientPath: 'node_modules/.prisma/client'` is working well after running the first `pnpm dev` with the dependencies installed. 
The only mistake that I see is that `pnpm install` is adding `prisma@5.7.1` to our `package.json` but out of that, everything is ok.

![Screenshot 2023-12-27 at 10 33 23](https://github.com/keystonejs/keystone/assets/9256107/6670f470-48d4-4591-b967-6b1e949fa59c)

---

About the monorepo, I did that, I forced the version `prisma@4.16.2` and it looks like is working well in all the versions, but I still need to test everything.

Thanks for the help! 🙌
